tareas/Tarea2/TareaVF2.jl

```diff
@@ -0,0 +1,680 @@
+#-
+
+#-
+#División
+ @testset "División de las estructuras duales" begin
+        @test Dual(1, 1)/ Dual(1, 1) == Dual(1.0,0.0)
+        @test 2 / -Dual(1.0, 1) == Dual(-2, 2)
+        @test -Dual(6.0, 2) / -3 == Dual(2.0, 2/3)        
+end;
+#-
+
+#-
+#Exponenciación
+ @testset "Exponenciación de las estructuras duales" begin
+        @test Dual(1, 1)^4 == Dual(1,4)
+        @test Dual(-1.0, 1)^4 == Dual(1.0,-4.0)
+end;
+#-
+
+#-
+#Equivalencia 
+ @testset "Equivalencia entre estructuras duales" begin 
+    @test Dual(1//4, 1//2) == Dual(0.25, 0.5)
+    @test Dual(1, 5) != Dual(5, 1)
+end;
+#-
+
+# $$f(x) = \frac{3x^2-8x+5}{7x^3-1}$$
+
+# - Evalúen la función `f(x) = (3x^2-8x+5)/(7x^3-1)` en el dual
+# `x₀ = 1 + \epsilon`, que representa la variable independiente
+# en el punto $x_0=1$. Rehagan este inciso usando un dual
+# en el punto $x_0=1$ usando aritmética de racionales.
+
+#-
+#Evaluamos f en el dual 1+\epsilon
+x=Dual(1,1)
+f=(3*x^2-8x+5)/(7*x^3-1)
+print(f)
+#-
+
+# #### Evaluamos analíticamente la función $f$ y su derivada para verificar el resultado anterior
+
+#-
+function funcionf(x)
+    (3*x^2-8x+5)/(7*x^3-1)
+end
+#-
+
+#-
+funcionf(1) 
+#-
+
+#-
+function derivadaf(x)
+    (8-6*x-105*x^2+112*x^3-21*x^4)/(-1+7*x^3)^2
+end
+#-
+
+#-
+derivadaf(1)
+#-
+
+# Por lo tanto tenermos que los resultados coinciden, ya que tenemos que la primera entra es 0 que corresponde con la función "funcionf" y la segunda entrada -0.333... coincide con la función "derivadaf", el resultado es consistente. 
+
+
+# #### Aritmética de racionales
+#-
+funcionf(Dual(1//1,1//1)) 
+#-
+
+# Notemos que el resultado es $\frac{-1}{3}$ y no -0.3333333333333333, es decir, es el valor exacto de la derivada y no una aproximación como en los casos anteriores.
+
+
+# $\textit{Ejercicio 3}$
+
+# - Recordando la regla de la cadena, extiendan el uso de `Dual` a 
+#las funciones elementales: `sqrt`, `exp`, `log`, `sin`, `cos`,
+# `tan`, `asin`, `acos`, `atan`, `sinh`, `cosh`, `tanh`, `asinh`,
+# `acosh`, `atanh`.
+#
+# - Muestren que su implementación da los resultados que se esperan 
+# usando pruebas como hicieron en ejercicios anteriores.
+# 
+# - Calculen la derivada de $h(x)=\sin(x^3−2/x^6)$ en $x_0=2$. ¿Qué 
+# tan preciso es el resultado? (Pueden usar cualquier otra manera 
+# de obtener el resultado correcto, sólo tienen que ser claros 
+# en la explicación.)
+#
+# - Grafiquen, para $x_0\in[1,5]$ la función $h^\prime(x)$.
+
+# #### Funciones elementales
+
+#-
+import Base: sqrt
+sqrt(D::Dual)= Dual(sqrt(fun(D)),(der(D)/(2*sqrt(fun(D)))))
+#-
+
+#-
+sqrt(Dual(4,1.0))
+#-
+
+#-
+import Base: exp
+exp(D::Dual)= Dual(exp(fun(D)),der(D)*exp(fun(D)))
+#-
+
+#-
+import Base: log
+log(D::Dual)= Dual(log(fun(D)),fun(D)/der(D))
+#-
+
+#-
+import Base: sin
+sin(D::Dual)= Dual(sin(fun(D)),cos(fun(D))*der(D))
+#-
+
+#-
+import Base: cos
+cos(D::Dual)= Dual(cos(fun(D)),-sin(fun(D))*der(D))
+#-
+
+#-
+import Base: tan
+tan(D::Dual)= Dual(tan(fun(D)),der(D)*(sec(fun(D)))**2)
```

Comment:
Al ejecutarlo, se genera el siguiente error: 
```
syntax: use "x^y" instead of "x**y" for exponentiation, and "x..." instead of "**x" for splatting.
```
Esto se soluciona cambiando `sec(fun(D)))**2`  por `sec(fun(D)))^2`